### PR TITLE
Disallow `VersionType.FORCE` versioning for 6.x indices

### PR DESCRIPTION
**Pull request by dakrone**
This is made up of two parts, a revert of b4cc3cd so `force` version is still available for 5.x indices, as well as a commit to disallow it for all 6.0+ indices.

Original commit message:

This was an error-prone version type that allowed overriding previous
version semantics. It could cause primaries and replicas to be out of
sync however, so it has been removed.

This is related to #20377, which removed the feature entirely. This
allows operations to continue to use the `force` version type if the
index was created before 6.0, in the event a document using it exists in
a translog being replayed.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/action/bulk/BulkWithUpdatesIT.java [hunk 1]
```diff
@@ -239,7 +239,6 @@ public void testBulkVersioning() throws Exception {
                         .setDoc("field", "3").setVersion(13).setVersionType(VersionType.INTERNAL))
                 .get();
 
-        assertThat(bulkResponse.getItems()[0].getFailureMessage(), containsString("version conflict"));
```
Comment (bleskes) [inline core/src/test/java/org/elasticsearch/action/bulk/BulkWithUpdatesIT.java]: this doesn't look like a rollback of https://github.com/elastic/elasticsearch/commit/b4cc3cd35dea2e3059142c6bd7e2eb13ca8944ac , are we sure it's ok?

**Review by bleskes — APPROVED** [review]
LGTM, left a minor sanity check

File: core/src/test/java/org/elasticsearch/action/bulk/BulkWithUpdatesIT.java (hunk 1 above)
Comment (dakrone) [inline core/src/test/java/org/elasticsearch/action/bulk/BulkWithUpdatesIT.java]: You're right, I've re-added the assert.

**Comment by dakrone** [comment]
Thanks for reviewing @bleskes!

**Pull request merged**